File: 2996-smallest-missing-integer-greater-than-sequential-prefix-sum/2996-smallest-missing-integer-greater-than-sequential-prefix-sum.cpp
class Solution {
public:
    int missingInteger(vector<int>& nums) {
        unordered_set<int> set;
        for(int i=0; i<nums.size(); i++){
            set.insert(nums[i]);
        }

        int ans = nums[0];
        for(int i=1; i<nums.size(); i++){
            if(nums[i] != nums[i-1] + 1) break;
            ans += nums[i];
        }

        while(set.count(ans)){
            ans++;
        }

        return ans;
    }
};

File: 2996-smallest-missing-integer-greater-than-sequential-prefix-sum/2996-smallest-missing-integer-greater-than-sequential-prefix-sum.java
class Solution {
    public int missingInteger(int[] nums) {
        HashSet<Integer> set = new HashSet<>();
        for(int i=0; i<nums.length; i++){
            set.add(nums[i]);
        }

        // int num = nums[0];
        int ans = nums[0];
        for(int i=1; i<nums.length; i++){
            if(nums[i] != nums[i-1] + 1) break;
            ans += nums[i];
        }

        while(set.contains(ans)){
            ans++;
        }

        return ans;
    }
}